### PR TITLE
Create scopes

### PR DESCRIPTION
**Pull request by kavishkafernando**
## Purpose
Create scopes for an API

### CODE REVIEW

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Scopes/CreateScope.jsx [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+                        />
+                    </APIPropertyField>
+                    <APIPropertyField name='Description'>
+                        <TextField
+                            style={{
+                                width: '100%',
+                            }}
+                            id='description'
+                            name='description'
+                            helperText='Short description about the scope'
```
Comment (pubudu538) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Scopes/CreateScope.jsx]: Shall we add i18n support for this as well?

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Scopes/CreateScope.jsx [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+        this.addScope = this.addScope.bind(this);
+        this.handleInputs = this.handleInputs.bind(this);
+    }
+
+    /**
+     * Add new scope
+     * @memberof Scopes
+     */
+    addScope() {
+        const hideMessage = message.loading('Adding the Scope ...', 0);
```
Comment (harsha89) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Scopes/CreateScope.jsx]: @kavishkafernando do we support for internationalization for these phrases? If not let's create a high priority github issue as we need to address it as soon as possible

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Scopes/CreateScope.jsx [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+        paddingLeft: theme.spacing.unit * 2.5,
+    },
+});
+
+/**
+ * Create new scoped for an API
+ * @class CreateScope
+ * @extends {Component}
+ */
+
```
Comment (npamudika) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Scopes/CreateScope.jsx]: Shall we remove this extra line?

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Scopes/CreateScope.jsx [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+                        color='primary'
+                        onClick={this.addScope}
+                        className={classes.buttonSave}
+                    >
+                        <FormattedMessage
+                            id='save'
+                            defaultMessage='Save'
+                        />
+                    </Button>
+                    <Link to={'/apis/' + this.props.api.id + '/scopes'}>
```
Comment (mushthaq33) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Scopes/CreateScope.jsx]: Shall we assign this to a variable and use it here?

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Scopes/Scopes.jsx [hunk 5]
```diff
@@ -189,12 +190,14 @@ class Scopes extends React.Component {
                                 </Typography>
                             </CardContent>
                             <CardActions>
-                                <Button variant='contained' color='primary' className={classes.button}>
-                                    <FormattedMessage
-                                        id='create.scopes'
-                                        defaultMessage='Create Scopes'
-                                    />
-                                </Button>
+                                <Link to={'/apis/' + api.id + '/scopes/create'}>
```
Comment (mushthaq33) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Scopes/Scopes.jsx]: Shall we assign this to a variable and use it here?

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Scopes/CreateScope.jsx [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+        marginTop: theme.spacing.unit * 10,
+    },
+    headline: {
+        paddingTop: theme.spacing.unit * 1.5,
+        paddingLeft: theme.spacing.unit * 2.5,
+    },
+});
+
+/**
+ * Create new scoped for an API
```
Comment (isharac) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Scopes/CreateScope.jsx]: typo 'scoped'

**Comment by kavishkafernando** [comment]
screen shot 2018-10-19 at 11 56 49 am

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Scopes/CreateScope.jsx [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+import TagsInput from 'react-tagsinput';
+import Api from 'AppData/api';
+import Button from '@material-ui/core/Button';
+import PropTypes from 'prop-types';
+import { withRouter } from 'react-router';
+import { Link } from 'react-router-dom';
+import { withStyles } from '@material-ui/core/styles';
+import Alert from 'AppComponents/Shared/Alert';
+
+
```
Comment (vithu30) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Scopes/CreateScope.jsx]: Shall we remove this extra line

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Scopes/CreateScope.jsx (hunk 6 above)
Comment (kavishkafernando) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Scopes/CreateScope.jsx]: fixed

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Scopes/CreateScope.jsx [hunk 8]
```diff
@@ -0,0 +1,207 @@
...
+                        onClick={this.addScope}
+                        className={classes.buttonSave}
+                    > Save
+                    </Button>
+                    <Link to={url}>
+                        <Button
+                            variant='contained'
+                            color='primary'
+                            className={classes.buttonCancel}
+                        >Cancel
```
[the remark this answers is not in the document]
Comment (kavishkafernando) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Scopes/CreateScope.jsx]: fixed

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Scopes/CreateScope.jsx [hunk 9]
```diff
@@ -0,0 +1,207 @@
...
+                            onChange={this.handleInputs}
+                            onlyUnique
+                        />
+                    </APIPropertyField>
+                    <Button
+                        variant='contained'
+                        color='primary'
+                        onClick={this.addScope}
+                        className={classes.buttonSave}
+                    > Save
```
[the remark this answers is not in the document]
Comment (kavishkafernando) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Scopes/CreateScope.jsx]: fixed

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Scopes/CreateScope.jsx (hunk 7 above)
Comment (kavishkafernando) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Scopes/CreateScope.jsx]: fixed

**Pull request merged**